README.md

```diff
@@ -50,13 +50,187 @@ module "postgresql" {
+                  }
+                }
+              }
+            ]
+          }
+EOF
+
+  iam_aurora_role_name = "AuroraRole-your-cluster" # ensure uniqueness of this one
+  iam_create_aurora_role = true
+  iam_auth_enabled = true
+  # ...
+}
+```
+
+### OpenSearch IRSA Role and Policy
+
+The OpenSearch module uses the following outputs from the EKS cluster module to define the IRSA role and policy:
+
+- `module.eks_cluster.oidc_provider_arn`: The ARN of the OIDC provider for the EKS cluster.
+- `module.eks_cluster.oidc_provider_id`: The ID of the OIDC provider for the EKS cluster.
+- `var.account_id`: Your account id
+- `var.opensearch_domain_name`: The name of the OpenSearch domain to access
+- `var.opensearch_namespace`: The namespace to allow access
+- `var.opensearch_service_account`: The ServiceAccount to allow access
+
+You need to define the IAM role trust policy and access policy for OpenSearch. Here's an example of how to define these policies using the outputs of the EKS cluster module:
+
+```hcl
+module "opensearch_domain" {
+  # ...
+  iam_create_opensearch_role = true
+  iam_opensearch_role_name = "OpenSearchRole-your-cluster" # ensure uniqueness of this one
+  iam_opensearch_access_policy = <<EOF
+            {
+              "Version": "2012-10-17",
+              "Statement": [
+                {
+                  "Effect": "Allow",
+                  "Action": [
+                    "es:ESHttpGet",
+                    "es:ESHttpPut",
+                    "es:ESHttpPost"
+                  ],
+                  "Resource": "arn:aws:es:${module.eks_cluster.region}:${var.account_id}:domain/${var.opensearch_domain_name}/*"
+                }
+              ]
+            }
+EOF
+
+  iam_role_trust_policy = <<EOF
+          {
+            "Version": "2012-10-17",
+            "Statement": [
+              {
+                "Effect": "Allow",
+                "Principal": {
+                  "Federated": "${module.eks_cluster.oidc_provider_arn}"
+                },
+                "Action": "sts:AssumeRoleWithWebIdentity",
+                "Condition": {
+                  "StringEquals": {
+                    "${module.eks_cluster.oidc_provider_id}:sub": "system:serviceaccount:${var.opensearch_namespace}:${var.opensearch_service_account}"
+                  }
+                }
+              }
+            ]
+          }
+EOF
+  # ...
+}
+```
+
+By defining the IRSA roles and policies using the outputs of the EKS cluster module, you can simplify the configuration and ensure that the roles and policies are created with the correct permissions and trust policies.
+
+Apply the Service Accounts definitions to your Kubernetes cluster:
+
+**Aurora Service Account**
+
+```yaml
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: aurora-service-account
+  namespace: <your-namespace>
+  annotations:
+    eks.amazonaws.com/role-arn: <arn:aws:iam:<YOUR-ACCOUNT-ID>:role/AuroraRole>
```

Comment:
I'd suggest somewhere in the text prior to mention the output where to pull the value from